src/BlockEntities/MobSpawnerEntity.h

```diff
@@ -33,46 +33,73 @@ class cMobSpawnerEntity :
 
-	/** Returns the entity type that will be spawn by this mob spawner. */
-	eMonsterType GetEntity(void) const { return m_Entity; }
-
-	/** Sets the entity type who will be spawn by this mob spawner. */
-	void SetEntity(eMonsterType a_EntityType) { m_Entity = a_EntityType; }
-
-	/** Returns the spawn delay. This is the tick delay that is needed to spawn new monsters. */
-	short GetSpawnDelay(void) const { return m_SpawnDelay; }
-
-	/** Sets the spawn delay. */
-	void SetSpawnDelay(short a_Delay) { m_SpawnDelay = a_Delay; }
-
-	/** Returns the amount of the nearby players in a 16-block radius. */
+	// Getters
+	int GetNearbyMonsterNum(eMonsterType a_EntityType);
 	int GetNearbyPlayersNum(void);
 
-	/** Returns the amount of this monster type in a 8-block radius (Y: 4-block radius). */
-	int GetNearbyMonsterNum(eMonsterType a_EntityType);
+	eMonsterType GetEntity(void) const       { return m_Entity; }
+	short GetSpawnCount(void) const          { return m_SpawnCount; }
+	short GetSpawnRange(void) const          { return m_SpawnRange; }
+	short GetSpawnDelay(void) const          { return m_SpawnDelay; }
+	short GetMinSpawnDelay(void) const       { return m_MinSpawnDelay; }
+	short GetMaxSpawnDelay(void) const       { return m_MaxSpawnDelay; }
+	short GetMaxNearbyEntities(void) const   { return m_MaxNearbyEntities; }
+	short GetRequiredPlayerRange(void) const { return m_RequiredPlayerRange; }
+
+	// Setters
+	void SetEntity(eMonsterType a_EntityType)                { m_Entity = a_EntityType; }
+	void SetSpawnDelay(short a_Delay)                        { m_SpawnDelay = a_Delay; }
+	void SetSpawnCount(short a_SpawnCount)                   { m_SpawnCount = a_SpawnCount; }
+	void SetSpawnRange(short a_SpawnRange);
+	void SetMinSpawnDelay(short a_Min)                       { m_MinSpawnDelay = a_Min; }
+	void SetMaxSpawnDelay(short a_Max)                       { m_MaxSpawnDelay = a_Max; }
+	void SetMaxNearbyEntities(short a_MaxNearbyEntities)     { m_MaxNearbyEntities = a_MaxNearbyEntities; }
+	void SetRequiredPlayerRange(short a_RequiredPlayerRange);
 
 	// tolua_end
 
-	static const char * GetClassStatic(void) { return "cMobSpawnerEntity"; }
-
 private:
 	/** The entity to spawn. */
 	eMonsterType m_Entity;
 
+	/** Time in ticks until the next entity spawns */
 	short m_SpawnDelay;
 
 	bool m_IsActive;
+
+	/** Number of entities the spawner tries to spawn each activation. */
+	short m_SpawnCount = 4;
+
+	/** Diameter of the square the new monsters are spawned in */
+	short m_SpawnRange = 8;
+
+	short m_MinSpawnDelay = 200;
+
+	short m_MaxSpawnDelay = 800;
+
+	/** Maximum amount of the same entity type in proximity. */
+	short m_MaxNearbyEntities = 6;
+
+	/** Maximum distance to player for activation */
+	short m_RequiredPlayerRange = 16;
+
+	/** The bounding boxes used to look for entities around the spawner.
+	Updated on updating m_RequiredPlayerRange or m_SpawnRange. */
+	cBoundingBox m_PlayerBoundingBox;
+	cBoundingBox m_EntityBoundingBox;
```

Comment:
You may be better off just creating the bounding box object on-demand. It's not expensive to construct and saves you from having to put the constructor calculation in multiple places.